include/dlr.h

```diff
@@ -47,6 +47,17 @@ typedef void (*DLRFreeFunctionPtr)(void*);
 typedef void* (*DLRMemalignFunctionPtr)(size_t, size_t);
 #endif
 
+#ifndef DLR_MODEL_ELEM
+#define DLR_MODEL_ELEM
+enum DLRModelElemType { HEXAGON_LIB, NEO_METADATA, TVM_GRAPH, TVM_LIB, TVM_PARAMS, RELAY_EXEC };
```

Comment:
I agree with sam that a string type would be easier to maintain/extend in future.

Reply:
Well, if we support new elem type we will need to document it in the CreateDLRModel function documentation. Anyway it will be code change in dlr.h, in cc files and in unit tests.
Enum is self-documented.
Inside the users application it should be possible to use new libdlr.so (included to the model) and older dlr.h. It should work in case they do not use newely added types in the application. If user want to use new type - they need to update dlr.h. But it's better to do it anyway - in order to get the documentation of the new type.

Reply:
why would it be a code change in dlr.h if we add a new file type string?

Reply:
Ok that makes sense, thanks.